tests/buffer/buffer_storage_common.h

```diff
@@ -39,53 +41,84 @@ template <typename alloc, typename T, int size, int dims>
 class buffer_storage_test {
  public:
   void operator()(util::logger &log, cl::sycl::range<dims> r) {
-    std::unique_ptr<T[]> data(new T[size]);
-    std::shared_ptr<T> data_shrd(new T[size], [](T *data) { delete[] data; });
+    auto deleter = std::default_delete<T[]>{};
```

Comment:
Not clear why you are defining a deleter if it is to use a default one at the end...

Reply:
After quite some time I realize the problem. You want `std::default_delete<T[]>{}` because `std::shared_ptr<T>` have `std::default_delete<T>{}` by default.
Non trivial to understand in uncommented code. :-(
But then, why not using `std::shared_ptr<T[]>` and `std::weak_ptr<T[]>` instead of lying?